### PR TITLE
tables: adding intel_me_info virtual table on Windows

### PR DESCRIPTION
**Pull request by muffins**
This ports the Intel ME virtual table to Windows. Sample output:
```
osquery> select * from intel_me_info;
+--------------+
| version      |
+--------------+
| 11.7.62976.3 |
+--------------+
```
Note: this value is wrong according to Intels detection tool. Will follow up shortly with a commit addressing this.

### CODE REVIEW

File: osquery/tables/system/windows/intel_me.cpp [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  interfaceData.cbSize = sizeof(SP_DEVICE_INTERFACE_DATA);
+
+  unsigned long cnt = 0;
+
+  auto ret = SetupDiEnumDeviceInterfaces(
+      deviceInfo, nullptr, guidPtr, cnt, &interfaceData);
+
+  std::string devPath;
+  while (ret == TRUE) {
+    // SP_DEVICE_INTERFACE_DATA stInterfaceData;
```
Comment (theopolis) [inline osquery/tables/system/windows/intel_me.cpp]: Nix this

File: osquery/tables/system/windows/intel_me.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            0x69,
+            0x15,
+            0xCE,
+            0x9B,
+            0xE5);
+
+struct me_version {
+  uint8_t major;
+  uint8_t minor;
+  uint16_t build;
```
Comment (theopolis) [inline osquery/tables/system/windows/intel_me.cpp]: As discussed, maybe flip build with hotfix? And flip the order below, so major.minor.hotfix.build?

**Comment by osqueryer** [comment]
:no_good: The commit 8e5fd39564c18bb47b3d4ace50e90d46f5a91ac7 (Job results: [2841](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2841/)) failed the code audit or documentation test.

**Comment by theopolis** [comment]
Can you flip the order in the Linux table too, so that it's reported, major.minor.hotfix.build

**Comment by theopolis** [comment]
Ah, will need to add the `linwin` to the API generator too:
```
Scanning dependencies of target docs
[100%] Generating API documentation with Doxygen/mkdocs
Traceback (most recent call last):
  File "/var/jenkins/workspace/osqueryPullRequestAudit/TargetSystem/aws-ubuntu-16.04/tools/codegen/genapi.py", line 306, in <module>
    main(len(sys.argv), sys.argv)
  File "/var/jenkins/workspace/osqueryPullRequestAudit/TargetSystem/aws-ubuntu-16.04/tools/codegen/genapi.py", line 281, in main
    api = gen_api(args.tables, profile)
  File "/var/jenkins/workspace/osqueryPullRequestAudit/TargetSystem/aws-ubuntu-16.04/tools/codegen/genapi.py", line 205, in gen_api
    platform_name = CANONICAL_PLATFORMS[platform]
KeyError: u'linwin'
```

**Comment by muffins** [comment]
Hrm, while that order seems more reasonable, it doesn't seem to be correct still:
```
osquery> select * from intel_me_info;
+----------+
| version  |
+----------+
| 11.7.0.3 |
+----------+
```
Vs
capture

**Comment by muffins** [comment]
Interseting... From my hardware manager in Windows, looking at the IME interface, I see yet another version, this one is much closer to what I'm seeing in osquery:
capture
*someone* is wrong here, just not sure whether it's me, windows, or intel ;) Now I'm wondering if we're not hitting the wrong interface, and Intel is gauging the vulnerability off of something different? Gunna look for different GUID values we might possibly consider...

**Comment by theopolis** [comment]
Ok so, that version response is just the SW version of the MEI client, right. We need to do more work, similar to the Linux implementation here.

1. Replace the `IOCTL_HECI_GET_VERSION` with `IOCTL_HECI_CONNECT_CLIENT`, which is `0x8000E004`.

2. Pass a GUID into the `DeviceIoControl` call, for the input-buff and input-buff-length. 

The GUID we need is:
```
DEFINE_GUID( MKHI_SUBSYSTEM_GUID, 0x8E6A6715, 0x9ABC, 0x4043, 0x88, 0xEF, 0x9E, 0x39, 0xC6, 0xF6, 0x3E, 0x0F );
```

So input-buffer is `(LPVOID)&MKHI_SUBSYSTEM_GUID`, size=16 bytes.

Then we need to write the handle with the command to get the FW version.

**Comment by theopolis** [comment]
On Linux the client then writes `0x0*4` and reads back a structure `mei_version` (much larger, but contains the fields we need). I'm not sure if it's the same API for this driver client.

```cpp
char cmd[4] = {0};
WriteFile(driver, cmd, 4, nullptr, nullptr);
char buffer[48] = {0};
ReadFile(driver, buffer, 20, nullptr, nullptr);
```

**Comment by theopolis** [comment]
The output buffer structure I referenced is:
```cpp
struct mei_version {
  uint32_t important_details[7];
  uint16_t major;
  uint16_t minor;
  uint16_t hotfix;
  uint16_t build;
  uint16_t r_major;
  uint16_t r_minor;
  uint16_t r_hotfix;
  uint16_t r_build;
  uint16_t codes[6];
};
```

**Comment by osqueryer** [comment]
:-1: The commit f35a26910daf74e9a5aec96d2c8ab5a611cd24cb (Job results: [3109](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3109/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit f35a26910daf74e9a5aec96d2c8ab5a611cd24cb (Job results: [6228](https://jenkins.osquery.io/job/osqueryPullRequestLinux/6228/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit f35a26910daf74e9a5aec96d2c8ab5a611cd24cb (Job results: [2328](https://jenkins.osquery.io/job/osqueryPullRequestOSX/2328/)) failed one or more tests (**macOS/OS X**).

**Comment by theopolis** [comment]
ok to test

**Comment by muffins** [comment]
This is looking good on test machines, merging.

**Pull request merged**